File: src/main/java/gestorAplicacion/instalaciones/InstalacionMenores.java
package gestorAplicacion.instalaciones;

//importar la clase cliente...
import gestorAplicacion.servicios.Registro;

import java.util.ArrayList;
import java.io.Serializable;

public class InstalacionMenores extends Instalacion implements Serializable {

    private static final long serialVersionUID = 1L;
    
    /*private ArrayList<Object> clientesQueSubieron = new ArrayList<>();
    private static ArrayList<Instalacion> instalacionesMenores = new ArrayList<>();*/
    
    
    public InstalacionMenores(String nombre,int edadRestriccion){
        super(nombre,edadRestriccion);

    }
    
    @Override
    public String agregarInstalacion(){
        if(!Registro.instalaciones.containsKey(this.getNombre())){
            Registro.instalaciones.put(this.getNombre(), this);
            return "Instalación agregada con éxito";
        }
        return "Ya existe una instalación con ese nombre";
    }
    
    
    @Override
    public String informacionInstalacion(){
        return "Nombre instalacion: "+getNombre()+
                "\nEdad restricción: "+getEdadRestriccion();
    }
    
    @Override
    public String tipoInstalacion(){
        return "Instalación para Menores de: 16 años";
    }
    
/*    public static void setInstalacionesMenores(ArrayList<Instalacion> im){
        instalacionesMenores = im;
    }
    
    public static ArrayList<Instalacion> getInstalacionesMenores(){
        return instalacionesMenores;
    }*/
    
}

File: practica2/src/gestorAplicacion/instalaciones/instalacionMenores.py
from .instalacion import Instalacion
from servicios.registro import Registro

class InstalacionMenores(Instalacion):

    def __init__(self, nombre):
        Instalacion.__init__(self,nombre, edadRestriccion= "Infante", costo = 50)
        self._MANTENIMIENTO = 300


    def agregarInstalacion(self):
        """metodo para agregar una nueva instalacion al parque"""
        if not (Registro.instalaciones.containsKey(self.getNombre())):
            Registro.instalaciones.put(self.getNombre(), self)
            return "Instalacion agregada con exito"
            
        return "ya exite una instalacion con ese nombre"

    def getCosto(self):
        return self._costo

    def informacionInstalacion(self):
        """metodo que muestra en pantalla quienes pueden entrar a la instalacion"""
        return f"Esta es una instalacion para menores de 16"
        
    def setEdadRestriccion (self, edadRestriccion):
        self._edadRestriccion = edadRestriccion
    
    
    def getEdadRestriccion(self):
        return self._edadRestriccion
        
    def setNombre(self, nombre):
        self._nombre = nombre
    
    def getNombre(self):
        return self._nombre

    def sumarUsoAntes(self):
        if self._MANTENIMIENTO > super().getUsosAntes():
            super().sumarUsosAntes()
            return True
        super().solicitarMantenimiento()
        return False

    def __str__(self):
        return f"Nombre: {self.getNombre()}, Restriccion: Menores,  Costo: 50, Mantenimiento: {super().getNecesitaMantenimiento()}"

